src/ui/component/filePicker/filePicker.tsx

```diff
@@ -0,0 +1,191 @@
+import { useCallback, useMemo, useState } from 'react'
+import type { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { DragAndDrop } from '@/ui/component/dragAndDrop/dragAndDrop'
+import type { DragDropState } from '@/ui/component/dragAndDrop/dragAndDrop'
+import { DropDownButton } from '@/ui/component/dropDownButton/dropDownButton'
+import type { Option } from '@/ui/component/dropDownButton/dropDownButton'
+import { Icon } from '@/ui/component/icon/icon'
+import { isError } from '@/util/util'
+import classNames from 'classnames'
+import './i18n/index'
+import './filePicker.scss'
+
+export type File = {
+  name: string
+  fullPath: string
+  size: number
+  mediaType: string
+  stream: ReadableStream<Uint8Array>
+}
+
+type FilePickerProps = {
+  onFileChange: (files: File[]) => void
+  hasFolderExplorer?: boolean
+  hasFileExplorer?: boolean
+  multiple?: boolean
+  onError?: (error: Error) => void
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const FilePicker: FC<FilePickerProps> = ({
+  onFileChange,
+  hasFolderExplorer = false,
+  hasFileExplorer = false,
+  multiple = false,
+  onError
+}): JSX.Element => {
+  const [isDraggingOver, setIsDraggingOver] = useState<boolean>(false)
+  const { t } = useTranslation('filePicker')
+
+  const handleDragAndDropState = useCallback((state: DragDropState) => {
+    state === 'dragging-over' ? setIsDraggingOver(true) : setIsDraggingOver(false)
+  }, [])
+
+  const toFileDTO = useCallback(
+    (file: globalThis.File) => ({
+      name: file.name,
+      fullPath: file.webkitRelativePath,
+      size: file.size,
+      mediaType: file.type,
+      stream: file.stream()
+    }),
+    []
+  )
+
+  const toFile = useCallback(
+    (item: DataTransferItem) => {
+      try {
+        const fsEntry = item.webkitGetAsEntry()
+        const file = item.getAsFile()
+
+        if (!fsEntry) throw new Error('Could not read item')
+        if (!file || fsEntry.isDirectory) throw new Error('Dropped item not a file')
+
+        return toFileDTO(file)
```

Comment:
That's not correct to map here the data transfer File from the DOM to a `DTO` structure because this is not generic. You cannot force the user of this component to handle this file type.

I would have expected to return a `File` list (as a `DOM` type) that will be handled by the parent and mapped as its convenience.

Let's keep it like that but we have to provide the changes in a very next future.